pyvo/mivot/utils/vocabulary.py

```diff
@@ -0,0 +1,155 @@
+    MODEL = namespace + "MODEL"
+    GLOBALS = namespace + "GLOBALS"
+    TEMPLATES = namespace + "TEMPLATES"
+    INSTANCE = namespace + "INSTANCE"
+    ATTRIBUTE = namespace + "ATTRIBUTE"
+    COLLECTION = namespace + "COLLECTION"
+    JOIN = namespace + "JOIN"
+    REFERENCE = namespace + "REFERENCE"
+    WHERE = namespace + "WHERE"
+    NOROLE = "NOROLE"
+
+
+@prototype_feature('MIVOT')
+class Att:
+    """
+    Constant used to identify attributes in MIVOT Element
+    """
+    dmrole = "dmrole"
+    dmtype = "dmtype"
+    dmid = "dmid"
+    name = "name"
+    value = "value"
+    dmref = "dmref"
+    tableref = "tableref"
+    sourceref = "sourceref"
+    ref = "ref"
+    primarykey = "PRIMARY_KEY"
+    foreignkey = "foreignkey"
+
+
+@prototype_feature('MIVOT')
+class MangoRoles:
+    """
+    Place holder for the MANGO draft roles
+    """
+    LONGITUDE = "longitude"
+    LATITUDE = "latitude"
+    PM_LONGITUDE = "pmLongitude"
+    PM_LATITUDE = "pmLatitude"
+    PARALLAX = "parallax"
+    RADIAL_VELOCITY = "radialVelocity"
+    EPOCH = "epoch"
+    FRAME = "frame"
+    EQUINOX = "equinox"
+    PMCOSDELTAPPLIED = "pmCosDeltApplied"
+
+
+EpochPropagation_fields = ["longitude", "latitude", "pmLongitude", "pmLatitude",
+                           "radialVelocity", "parallax", "epoch", "frame", "equinox"]
+
+
+skycoord_param_default = {
+    MangoRoles.LONGITUDE: 'ra', MangoRoles.LATITUDE: 'dec', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_ra_cosdec', MangoRoles.PM_LATITUDE: 'pm_dec',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime',
+    MangoRoles.FRAME: 'frame'}
+
+skycoord_param_fk4_fk5 = {
+    MangoRoles.LONGITUDE: 'ra', MangoRoles.LATITUDE: 'dec', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_ra_cosdec', MangoRoles.PM_LATITUDE: 'pm_dec',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime',
+    MangoRoles.FRAME: 'frame', MangoRoles.EQUINOX: 'equinox'}
+
+skycoord_param_galactic = {
+    MangoRoles.LONGITUDE: 'l', MangoRoles.LATITUDE: 'b', MangoRoles.PARALLAX: 'distance',
+    MangoRoles.PM_LONGITUDE: 'pm_l_cosb', MangoRoles.PM_LATITUDE: 'pm_b',
+    MangoRoles.RADIAL_VELOCITY: 'radial_velocity', MangoRoles.EPOCH: 'obstime',
+    MangoRoles.FRAME: 'frame'}
+
+
+@prototype_feature('MIVOT')
+def key_match(searched_key, key_set):
+    """
+    Check if any key in the key_set starts with the searched_key.
+    Parameters
+    ----------
+    searched_key : str
+        The key to search for.
+    key_set : str, list, or odict_keys
+        The set of keys to check.
+    Returns
+    -------
+    str or None
+        The matched key if found, otherwise None.
+    """
+    if isinstance(key_set, str):
+        return key_set.startswith(searched_key)
+    else:
+        # May be a list or an odict_keys
+        for key in key_set:
+            if key.startswith(searched_key) is True:
+                return key
+    return None
```

Comment:
Is any of this used or needed?

Reply:
Unused code removed from vocabulary.py (c24931f782c1092253511548b506d5f3abf7bd78). 
The unit question will be addressed with tye issue #532.